From 3a1ee2ea6a6c12c101d3fba14c39f4a748d2f9c8 Mon Sep 17 00:00:00 2001
From: flar2 <asegaert@gmail.com>
Date: Mon, 11 Nov 2013 00:42:12 -0500
Subject: [PATCH 032/276] More overclocking options


Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 arch/arm/mach-msm/acpuclock-8974.c |   50 ++++++++++++++++++++++++++++++++++++
 1 file changed, 50 insertions(+)

diff --git a/arch/arm/mach-msm/acpuclock-8974.c b/arch/arm/mach-msm/acpuclock-8974.c
index 933bd0e..b436816 100644
--- a/arch/arm/mach-msm/acpuclock-8974.c
+++ b/arch/arm/mach-msm/acpuclock-8974.c
@@ -310,6 +310,29 @@ static struct l2_level l2_freq_tbl_v2_elementalx[] __initdata = {
 	{ }
 };
 
+static struct l2_level l2_freq_tbl_v2_ultra[] __initdata = {
+	[0]  = { {  300000, PLL_0, 0,   0 }, LVL_LOW,   950000, 0 },
+	[1]  = { {  345600, HFPLL, 2,  36 }, LVL_LOW,   950000, 1 },
+	[2]  = { {  422400, HFPLL, 2,  44 }, LVL_LOW,   950000, 2 },
+	[3]  = { {  499200, HFPLL, 2,  52 }, LVL_LOW,   950000, 3 },
+	[4]  = { {  576000, HFPLL, 1,  30 }, LVL_LOW,   950000, 4 },
+	[5]  = { {  652800, HFPLL, 1,  34 }, LVL_NOM,   950000, 4 },
+	[6]  = { {  729600, HFPLL, 1,  38 }, LVL_NOM,   950000, 4 },
+	[7]  = { {  806400, HFPLL, 1,  42 }, LVL_NOM,   950000, 4 },
+	[8]  = { {  883200, HFPLL, 1,  46 }, LVL_NOM,   950000, 5 },
+	[9]  = { {  960000, HFPLL, 1,  50 }, LVL_NOM,   950000, 5 },
+	[10] = { { 1036800, HFPLL, 1,  54 }, LVL_NOM,   950000, 5 },
+	[11] = { { 1113600, HFPLL, 1,  58 }, LVL_HIGH, 1050000, 6 },
+	[12] = { { 1190400, HFPLL, 1,  62 }, LVL_HIGH, 1050000, 6 },
+	[13] = { { 1267200, HFPLL, 1,  66 }, LVL_HIGH, 1050000, 6 },
+	[14] = { { 1344000, HFPLL, 1,  70 }, LVL_HIGH, 1050000, 6 },
+	[15] = { { 1420800, HFPLL, 1,  74 }, LVL_HIGH, 1050000, 6 },
+	[16] = { { 1497600, HFPLL, 1,  78 }, LVL_HIGH, 1050000, 6 },
+	[17] = { { 1574400, HFPLL, 1,  82 }, LVL_HIGH, 1050000, 7 },
+	[18] = { { 1651200, HFPLL, 1,  86 }, LVL_HIGH, 1050000, 7 },
+	[19] = { { 1920000, HFPLL, 1, 100 }, LVL_HIGH, 1050000, 9 },
+	{ }
+};
 static struct acpu_level acpu_freq_tbl_2g_pvs0[] __initdata = {
 	{ 1, {  300000, PLL_0, 0,   0 },  L2(0),  815000,  73 },
 	{ 0, {  345600, HFPLL, 2,  36 },  L2(1),  825000,  85 },
@@ -743,6 +766,9 @@ static struct acpu_level acpu_freq_tbl_2p3g_pvs0[] __initdata = {
 	{ 1, { 2572800, HFPLL, 1, 134 }, L2(19), 1160000, 784 },
 	{ 1, { 2649600, HFPLL, 1, 138 }, L2(19), 1175000, 808 },
 	{ 1, { 2726400, HFPLL, 1, 142 }, L2(19), 1195000, 831 },
+	{ 1, { 2803200, HFPLL, 1, 146 }, L2(19), 1195000, 854 },
+	{ 1, { 2880000, HFPLL, 1, 150 }, L2(19), 1195000, 876 },
+	{ 1, { 2956800, HFPLL, 1, 154 }, L2(19), 1195000, 897 },
 	{ 0, { 0 } }
 };
 
@@ -780,6 +806,9 @@ static struct acpu_level acpu_freq_tbl_2p3g_pvs1[] __initdata = {
 	{ 1, { 2572800, HFPLL, 1, 134 }, L2(19), 1135000, 784 },
 	{ 1, { 2649600, HFPLL, 1, 138 }, L2(19), 1150000, 808 },
 	{ 1, { 2726400, HFPLL, 1, 142 }, L2(19), 1165000, 831 },
+	{ 1, { 2803200, HFPLL, 1, 146 }, L2(19), 1180000, 854 },
+	{ 1, { 2880000, HFPLL, 1, 150 }, L2(19), 1195000, 876 },
+	{ 1, { 2956800, HFPLL, 1, 154 }, L2(19), 1195000, 897 },
 	{ 0, { 0 } }
 };
 
@@ -817,6 +846,9 @@ static struct acpu_level acpu_freq_tbl_2p3g_pvs2[] __initdata = {
 	{ 1, { 2572800, HFPLL, 1, 134 }, L2(19), 1110000, 784 },
 	{ 1, { 2649600, HFPLL, 1, 138 }, L2(19), 1125000, 808 },
 	{ 1, { 2726400, HFPLL, 1, 142 }, L2(19), 1140000, 831 },
+	{ 1, { 2803200, HFPLL, 1, 146 }, L2(19), 1165000, 854 },
+	{ 1, { 2880000, HFPLL, 1, 150 }, L2(19), 1180000, 876 },
+	{ 1, { 2956800, HFPLL, 1, 154 }, L2(19), 1195000, 897 },
 	{ 0, { 0 } }
 };
 
@@ -854,6 +886,9 @@ static struct acpu_level acpu_freq_tbl_2p3g_pvs3[] __initdata = {
 	{ 1, { 2572800, HFPLL, 1, 134 }, L2(19), 1085000, 784 },
 	{ 1, { 2649600, HFPLL, 1, 138 }, L2(19), 1100000, 808 },
 	{ 1, { 2726400, HFPLL, 1, 142 }, L2(19), 1115000, 831 },
+	{ 1, { 2803200, HFPLL, 1, 146 }, L2(19), 1130000, 854 },
+	{ 1, { 2880000, HFPLL, 1, 150 }, L2(19), 1145000, 876 },
+	{ 1, { 2956800, HFPLL, 1, 154 }, L2(19), 1160000, 897 },
 	{ 0, { 0 } }
 };
 
@@ -891,6 +926,9 @@ static struct acpu_level acpu_freq_tbl_2p3g_pvs4[] __initdata = {
 	{ 1, { 2572800, HFPLL, 1, 134 }, L2(19), 1060000, 784 },
 	{ 1, { 2649600, HFPLL, 1, 138 }, L2(19), 1075000, 808 },
 	{ 1, { 2726400, HFPLL, 1, 142 }, L2(19), 1090000, 831 },
+	{ 1, { 2803200, HFPLL, 1, 146 }, L2(19), 1105000, 854 },
+	{ 1, { 2880000, HFPLL, 1, 150 }, L2(19), 1120000, 876 },
+	{ 1, { 2956800, HFPLL, 1, 154 }, L2(19), 1135000, 897 },
 	{ 0, { 0 } }
 };
 
@@ -928,6 +966,9 @@ static struct acpu_level acpu_freq_tbl_2p3g_pvs5[] __initdata = {
 	{ 1, { 2572800, HFPLL, 1, 134 }, L2(19), 1035000, 784 },
 	{ 1, { 2649600, HFPLL, 1, 138 }, L2(19), 1050000, 808 },
 	{ 1, { 2726400, HFPLL, 1, 142 }, L2(19), 1065000, 831 },
+	{ 1, { 2803200, HFPLL, 1, 146 }, L2(19), 1080000, 854 },
+	{ 1, { 2880000, HFPLL, 1, 150 }, L2(19), 1095000, 876 },
+	{ 1, { 2956800, HFPLL, 1, 154 }, L2(19), 1110000, 897 },
 	{ 0, { 0 } }
 };
 
@@ -965,6 +1006,9 @@ static struct acpu_level acpu_freq_tbl_2p3g_pvs6[] __initdata = {
 	{ 1, { 2572800, HFPLL, 1, 134 }, L2(19), 1005000, 784 },
 	{ 1, { 2649600, HFPLL, 1, 138 }, L2(19), 1020000, 808 },
 	{ 1, { 2726400, HFPLL, 1, 142 }, L2(19), 1035000, 831 },
+	{ 1, { 2803200, HFPLL, 1, 146 }, L2(19), 1050000, 854 },
+	{ 1, { 2880000, HFPLL, 1, 150 }, L2(19), 1065000, 876 },
+	{ 1, { 2956800, HFPLL, 1, 154 }, L2(19), 1080000, 897 },
 	{ 0, { 0 } }
 };
 
@@ -1036,6 +1080,8 @@ static int __init get_opt_level(char *l2_opt)
 		opt_bin = 0;
 	} else if (strcmp(l2_opt, "1") == 0) {
 		opt_bin = 1;
+	} else if (strcmp(l2_opt, "2") == 0) {
+		opt_bin = 2;
 	} else {
 		opt_bin = 0;
 	}
@@ -1087,6 +1133,10 @@ static int __init acpuclk_8974_probe(struct platform_device *pdev)
 		acpuclk_8974_params.l2_freq_tbl = l2_freq_tbl_v2_elementalx;
 		acpuclk_8974_params.l2_freq_tbl_size = sizeof(l2_freq_tbl_v2_elementalx);
 	}
+	if (opt_bin == 2) {
+		acpuclk_8974_params.l2_freq_tbl = l2_freq_tbl_v2_ultra;
+		acpuclk_8974_params.l2_freq_tbl_size = sizeof(l2_freq_tbl_v2_ultra);
+	}
 
 	return acpuclk_krait_init(&pdev->dev, &acpuclk_8974_params);
 }
-- 
1.7.9.5

