From 1a9b96fbb45c12e824d99f4997dfc3e3216f1694 Mon Sep 17 00:00:00 2001
From: flar2 <asegaert@gmail.com>
Date: Mon, 18 Nov 2013 20:16:02 -0500
Subject: [PATCH 072/276] Change default max screen off frequency to 1190400


Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 arch/arm/mach-msm/cpufreq.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-msm/cpufreq.c b/arch/arm/mach-msm/cpufreq.c
index 8698927..c51d232 100644
--- a/arch/arm/mach-msm/cpufreq.c
+++ b/arch/arm/mach-msm/cpufreq.c
@@ -45,7 +45,7 @@ static DEFINE_PER_CPU(struct cpufreq_work_struct, cpufreq_work);
 static struct workqueue_struct *msm_cpufreq_wq;
 
 /* maxscroff */
-uint32_t maxscroff_freq = 729600;
+uint32_t maxscroff_freq = 1190400;
 uint32_t maxscroff = 0; 
 
 struct cpufreq_suspend_t {
-- 
1.7.9.5

