From 8e6695496d298f5936f6d3ba3eb7b70b17071035 Mon Sep 17 00:00:00 2001
From: flar2 <asegaert@gmail.com>
Date: Sat, 21 Dec 2013 02:09:48 -0500
Subject: [PATCH 231/276] cpufreq_elementalx:  add counter for more gradual
 boosting


Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 drivers/cpufreq/cpufreq_elementalx.c |   20 ++++++++++++++++----
 1 file changed, 16 insertions(+), 4 deletions(-)

diff --git a/drivers/cpufreq/cpufreq_elementalx.c b/drivers/cpufreq/cpufreq_elementalx.c
index 3cca32f..60b6cf5 100644
--- a/drivers/cpufreq/cpufreq_elementalx.c
+++ b/drivers/cpufreq/cpufreq_elementalx.c
@@ -34,6 +34,7 @@
 //gboost
 #include <mach/kgsl.h>
 static int orig_up_threshold = 90;
+static int g_count = 0;
 
 #define DEF_SAMPLING_RATE			(30000)
 #define DEF_FREQUENCY_DOWN_DIFFERENTIAL		(10)
@@ -1242,15 +1243,26 @@ if (dbs_tuners_ins.gboost) {
 }
 
 //graphics boost
-	if (graphics_boost < 4 && dbs_tuners_ins.gboost) {
+if (dbs_tuners_ins.gboost) {
+
+	if (g_count < 100 && graphics_boost < 5) {
+		++g_count;
+	} else if (g_count > 1) {
+		--g_count;
+		--g_count;
+	}
+
+	if (graphics_boost < 4 && g_count > 80) {
 		dbs_tuners_ins.up_threshold = 60 + (graphics_boost * 10);
 	} else {
 		dbs_tuners_ins.up_threshold = orig_up_threshold;
 	}
-	if ((graphics_boost == 3 || graphics_boost == 4) && dbs_tuners_ins.gboost) {
+	if (g_count > 40) {
 		input_event_boost = true;
 		input_event_boost_expired = jiffies + usecs_to_jiffies(dbs_tuners_ins.sampling_rate * 2);
 	}
+
+}
 //end
 
 	if (num_online_cpus() > 1) {
@@ -1573,8 +1585,8 @@ static int cpufreq_governor_dbs(struct cpufreq_policy *policy,
 		this_dbs_info->cpu = cpu;
 		this_dbs_info->rate_mult = 1;
 		elementalx_powersave_bias_init_cpu(cpu);
-		set_two_phase_freq(1574000);
-	        set_input_event_min_freq_by_cpu(1, 1190000);
+		set_two_phase_freq(1267200);
+	        set_input_event_min_freq_by_cpu(1, 1267200);
         	set_input_event_min_freq_by_cpu(2, 1036000);
         	set_input_event_min_freq_by_cpu(3, 729000);
         	set_input_event_min_freq_by_cpu(4, 729000);
-- 
1.7.9.5

