From 162fd12453311a24c4db1f91b8916bd6737aab1c Mon Sep 17 00:00:00 2001
From: flar2 <asegaert@gmail.com>
Date: Thu, 26 Dec 2013 01:19:46 -0500
Subject: [PATCH 239/276] cpufreq_elementalx: very slight boost


Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 drivers/cpufreq/cpufreq_elementalx.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/cpufreq/cpufreq_elementalx.c b/drivers/cpufreq/cpufreq_elementalx.c
index 60b6cf5..f5ffadf 100644
--- a/drivers/cpufreq/cpufreq_elementalx.c
+++ b/drivers/cpufreq/cpufreq_elementalx.c
@@ -1257,7 +1257,7 @@ if (dbs_tuners_ins.gboost) {
 	} else {
 		dbs_tuners_ins.up_threshold = orig_up_threshold;
 	}
-	if (g_count > 40) {
+	if (g_count > 35) {
 		input_event_boost = true;
 		input_event_boost_expired = jiffies + usecs_to_jiffies(dbs_tuners_ins.sampling_rate * 2);
 	}
-- 
1.7.9.5

