From 973162a9c80ecb4e2631303697503c4aa15f2010 Mon Sep 17 00:00:00 2001
From: flar2 <asegaert@gmail.com>
Date: Sat, 28 Dec 2013 09:47:18 -0500
Subject: [PATCH 246/276] Revert "Revert "fs: sync: enable fsync by default.""

This reverts commit 56809d9b92d96413a536896c8d02550a0a2774c0.

Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 fs/sync.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/sync.c b/fs/sync.c
index 3c506d8..5162fbe 100644
--- a/fs/sync.c
+++ b/fs/sync.c
@@ -18,7 +18,7 @@
 #include <linux/backing-dev.h>
 #include "internal.h"
 
-bool fsync_enabled = false;
+bool fsync_enabled = true;
 module_param(fsync_enabled, bool, 0755);
 
 #define VALID_FLAGS (SYNC_FILE_RANGE_WAIT_BEFORE|SYNC_FILE_RANGE_WRITE| \
-- 
1.7.9.5

