From 3937c002e64ad42744c5c80763599c82e5cff1bb Mon Sep 17 00:00:00 2001
From: flar2 <asegaert@gmail.com>
Date: Wed, 1 Jan 2014 15:27:09 -0500
Subject: [PATCH 271/276] [doubletap2wake]: fullscreen option


Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 drivers/input/touchscreen/doubletap2wake.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/input/touchscreen/doubletap2wake.c b/drivers/input/touchscreen/doubletap2wake.c
index 1b6d159..84ab106 100644
--- a/drivers/input/touchscreen/doubletap2wake.c
+++ b/drivers/input/touchscreen/doubletap2wake.c
@@ -84,6 +84,9 @@ static int __init read_dt2w_cmdline(char *dt2w)
 	if (strcmp(dt2w, "1") == 0) {
 		pr_info("[cmdline_dt2w]: DoubleTap2Wake enabled. | dt2w='%s'\n", dt2w);
 		dt2w_switch = 1;
+	} else if (strcmp(dt2w, "2") == 0) {
+		pr_info("[cmdline_dt2w]: DoubleTap2Wake fullscreen enabled. | dt2w='%s'\n", dt2w);
+		dt2w_switch = 2;
 	} else if (strcmp(dt2w, "0") == 0) {
 		pr_info("[cmdline_dt2w]: DoubleTap2Wake disabled. | dt2w='%s'\n", dt2w);
 		dt2w_switch = 0;
@@ -152,6 +155,9 @@ static void detect_doubletap2wake(int x, int y, bool st)
 	if (x < 100 || x > 980)
         	return;
 
+	if (dt2w_switch < 2 && y < 1000)
+        	return;
+
 	if ((single_touch) && (dt2w_switch > 0) && (exec_count) && (touch_cnt)) {
 		touch_cnt = false;
 		if (touch_nr == 0) {
-- 
1.7.9.5

