From 1784c9e4654b9f55aa48cfe1b98ef8007bbee476 Mon Sep 17 00:00:00 2001
From: Dennis Rassmann <showp1984@gmail.com>
Date: Fri, 21 Dec 2012 05:27:34 +0100
Subject: [PATCH 164/338] msm_mpdecision: on resume pull up cpu1

Signed-off-by: Dennis Rassmann <showp1984@gmail.com>
Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 arch/arm/mach-msm/msm_mpdecision.c |   13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/arm/mach-msm/msm_mpdecision.c b/arch/arm/mach-msm/msm_mpdecision.c
index 640cfb5..25a7ac7 100644
--- a/arch/arm/mach-msm/msm_mpdecision.c
+++ b/arch/arm/mach-msm/msm_mpdecision.c
@@ -317,7 +317,20 @@ static void msm_mpdec_resume(void)
 	for_each_possible_cpu(cpu)
 		per_cpu(msm_mpdec_cpudata, cpu).device_suspended = false;
 
+	mutex_lock(&per_cpu(msm_mpdec_cpudata, 1).suspend_mutex);
+	if (!cpu_online(1)) {
+		/* Always enable cpu1 when screen comes online.
+		 * This boosts the wakeup process. */
+		cpu_up(1);
+		per_cpu(msm_mpdec_cpudata, 1).on_time = ktime_to_ms(ktime_get());
+		per_cpu(msm_mpdec_cpudata, 1).online = true;
+		pr_info(MPDEC_TAG"Screen -> on. Hot plugged CPU1 | Mask=[%d%d%d%d]\n",
+                        cpu_online(0), cpu_online(1), cpu_online(2), cpu_online(3));
+	}
+	mutex_unlock(&per_cpu(msm_mpdec_cpudata, 1).suspend_mutex);
+
         /* wake up main work thread */
+        was_paused = true;
         queue_delayed_work(msm_mpdec_workq, &msm_mpdec_work, 0);
 
         pr_info(MPDEC_TAG"Screen -> on. Activated mpdecision. | Mask=[%d%d%d%d]\n",
-- 
1.7.9.5

