From 8f36c7a1792b82211979bedba747331f79ee2573 Mon Sep 17 00:00:00 2001
From: faux123 <reioux@gmail.com>
Date: Wed, 5 Dec 2012 06:19:54 -0800
Subject: [PATCH 021/338] intellidemand: increase boost duration to 2.5 seconds
 from 1.5

Signed-off-by: faux123 <reioux@gmail.com>
---
 drivers/cpufreq/cpufreq_intellidemand.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/cpufreq/cpufreq_intellidemand.c b/drivers/cpufreq/cpufreq_intellidemand.c
index 6ee01a8..9c27973 100644
--- a/drivers/cpufreq/cpufreq_intellidemand.c
+++ b/drivers/cpufreq/cpufreq_intellidemand.c
@@ -49,7 +49,7 @@
 #define MIN_FREQUENCY_UP_THRESHOLD		(11)
 #define MAX_FREQUENCY_UP_THRESHOLD		(100)
 #define MIN_FREQUENCY_DOWN_DIFFERENTIAL		(1)
-#define DEFAULT_FREQ_BOOST_TIME			(1500000)
+#define DEFAULT_FREQ_BOOST_TIME			(2500000)
 #define DEF_SAMPLING_RATE			(50000)
 #define BOOSTED_SAMPLING_RATE			(15000)
 #define DBS_INPUT_EVENT_MIN_FREQ		(1026000)
-- 
1.8.1.2

