From e9489635f9fc082995e92aabc429de906740ec12 Mon Sep 17 00:00:00 2001
From: Paul Reioux <reioux@gmail.com>
Date: Thu, 23 May 2013 22:07:04 -0500
Subject: [PATCH 147/507] intellidemand: enable sec_touchscreen input boosting

Signed-off-by: Paul Reioux <reioux@gmail.com>
Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 drivers/cpufreq/cpufreq_intellidemand.c |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/cpufreq/cpufreq_intellidemand.c b/drivers/cpufreq/cpufreq_intellidemand.c
index bd4a639..dfe6d95 100644
--- a/drivers/cpufreq/cpufreq_intellidemand.c
+++ b/drivers/cpufreq/cpufreq_intellidemand.c
@@ -1676,7 +1676,7 @@ static void dbs_refresh_callback(struct work_struct *work)
 	}
 
 	if (policy->cur < DBS_INPUT_EVENT_MIN_FREQ) {
-#if 0
+#if 1
 		pr_info("%s: set cpufreq to DBS_INPUT_EVENT_MIN_FREQ(%d) \
 			directly due to input events!\n", __func__, \
 			DBS_INPUT_EVENT_MIN_FREQ);
@@ -1700,6 +1700,7 @@ bail_acq_sema_failed:
 }
 
 static unsigned int enable_dbs_input_event = 1;
+
 static void dbs_input_event(struct input_handle *handle, unsigned int type,
 		unsigned int code, int value)
 {
@@ -1719,6 +1720,9 @@ static void dbs_input_event(struct input_handle *handle, unsigned int type,
 			sampling_rate_boosted = 1;
 		}
 
+		/* debug mesg */
+		//pr_info("screen touched!\n");
+
 		for_each_online_cpu(i)
 			queue_work_on(i, input_wq, &per_cpu(dbs_refresh_work, i).work);
 	}
@@ -1728,6 +1732,7 @@ static void dbs_input_event(struct input_handle *handle, unsigned int type,
 static int input_dev_filter(const char *input_dev_name)
 {
 	if (strstr(input_dev_name, "touchscreen") ||
+		strstr(input_dev_name, "sec_touchscreen") ||
 		strstr(input_dev_name, "-keypad") ||
 		strstr(input_dev_name, "-nav") ||
 		strstr(input_dev_name, "-oj")) {
@@ -1766,6 +1771,7 @@ static int dbs_input_connect(struct input_handler *handler,
 	if (error)
 		goto err1;
 
+	pr_info("%s found and connected!\n", dev->name);
 	return 0;
 err1:
 	input_unregister_handle(handle);
-- 
1.7.9.5

