From 1fc41477c23c1af8f53878cd4b4c52ad42106e8d Mon Sep 17 00:00:00 2001
From: Paul Reioux <reioux@gmail.com>
Date: Tue, 4 Jun 2013 02:36:19 -0500
Subject: [PATCH 192/507] Intellidemand: limit min cpu perflock to boot cpu
 only

Signed-off-by: Paul Reioux <reioux@gmail.com>
Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 drivers/cpufreq/cpufreq_intellidemand.c |   14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/drivers/cpufreq/cpufreq_intellidemand.c b/drivers/cpufreq/cpufreq_intellidemand.c
index 5b00ce0..6241d58 100644
--- a/drivers/cpufreq/cpufreq_intellidemand.c
+++ b/drivers/cpufreq/cpufreq_intellidemand.c
@@ -1353,12 +1353,14 @@ static void do_dbs_timer(struct work_struct *work)
 			rq_persist_count--;
 
 #ifdef CONFIG_CPUFREQ_ID_PERFLOCK
-	if (num_online_cpus() >= 2) {
-		if (saved_policy_min != 0)
-			policy->min = saved_policy_min;
-	} else if (num_online_cpus() == 1) {
-		saved_policy_min = policy->min;
-		policy->min = DBS_PERFLOCK_MIN_FREQ;
+	if (cpu == 0) {
+		if (num_online_cpus() >= 2) {
+			if (saved_policy_min != 0)
+				policy->min = saved_policy_min;
+		} else if (num_online_cpus() == 1) {
+			saved_policy_min = policy->min;
+			policy->min = DBS_PERFLOCK_MIN_FREQ;
+		}
 	}
 #endif
 
-- 
1.7.9.5

