From 244b2961dd81ad5154818c426b1ddd34aab1d5a8 Mon Sep 17 00:00:00 2001
From: Paul Reioux <reioux@gmail.com>
Date: Sun, 27 Oct 2013 03:18:39 -0500
Subject: [PATCH 348/507] include/scsi/scsi.h: remove obsolete function

Signed-off-by: Paul Reioux <reioux@gmail.com>
Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 include/scsi/scsi.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/scsi/scsi.h b/include/scsi/scsi.h
index cc37cd8..a1ab569 100644
--- a/include/scsi/scsi.h
+++ b/include/scsi/scsi.h
@@ -562,6 +562,6 @@ static inline __u32 scsi_to_u32(__u8 *ptr)
 	return (ptr[0]<<24) + (ptr[1]<<16) + (ptr[2]<<8) + ptr[3];
 }
 
-struct scsi_disk *scsi_disk_get_from_dev(struct device *dev);
+//struct scsi_disk *scsi_disk_get_from_dev(struct device *dev);
 
 #endif /* _SCSI_SCSI_H */
-- 
1.7.9.5

