From a3f82f48776feb0b3c246266c47ac62bdd2d327c Mon Sep 17 00:00:00 2001
From: Paul Reioux <reioux@gmail.com>
Date: Sun, 29 Dec 2013 21:43:25 -0800
Subject: [PATCH 499/507] zswap: use LZ4 if available

Signed-off-by: Paul Reioux <reioux@gmail.com>
Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 mm/zswap.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/mm/zswap.c b/mm/zswap.c
index 50bf537..5af53ca 100644
--- a/mm/zswap.c
+++ b/mm/zswap.c
@@ -76,7 +76,11 @@ static bool zswap_enabled = 1;
 module_param_named(enabled, zswap_enabled, bool, 0);
 
 /* Compressor to be used by zswap (fixed at boot for now) */
+#ifdef CONFIG_CRYPTO_LZ4
+#define ZSWAP_COMPRESSOR_DEFAULT "lz4"
+#else
 #define ZSWAP_COMPRESSOR_DEFAULT "lzo"
+#endif
 static char *zswap_compressor = ZSWAP_COMPRESSOR_DEFAULT;
 module_param_named(compressor, zswap_compressor, charp, 0);
 
-- 
1.7.9.5

