From 2e98aaead9ae8fc83ad0eaad2efbe061b98d79a5 Mon Sep 17 00:00:00 2001
From: Francisco Franco <franciscofranco.1990@gmail.com>
Date: Tue, 12 Nov 2013 19:39:27 +0000
Subject: [PATCH 064/517] drivers: thermal: fix throttling behavior.

Signed-off-by: Francisco Franco <franciscofranco.1990@gmail.com>
Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 drivers/thermal/msm_thermal.c |   10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/drivers/thermal/msm_thermal.c b/drivers/thermal/msm_thermal.c
index e999acd..d0fc2d0 100644
--- a/drivers/thermal/msm_thermal.c
+++ b/drivers/thermal/msm_thermal.c
@@ -40,7 +40,7 @@ static struct cpus {
  */
 unsigned int polling = HZ*2;
 
-unsigned int temp_threshold = 70;
+unsigned int temp_threshold = 75;
 module_param(temp_threshold, int, 0755);
 
 static struct msm_thermal_data msm_thermal_info;
@@ -67,19 +67,23 @@ static int update_cpu_max_freq(int cpu, uint32_t max_freq)
 		pr_info("%s: Limiting cpu%d max frequency to %d\n",
 				KBUILD_MODNAME, cpu, max_freq);
 	} else {
-		cpu_stats.throttling = false;
+		cpu_stats.throttling = false; 
 		pr_info("%s: Max frequency reset for cpu%d\n",
 				KBUILD_MODNAME, cpu);
 	}
 
+	get_online_cpus();
 	if (cpu_online(cpu)) {
 		struct cpufreq_policy *policy = cpufreq_cpu_get(cpu);
 		if (!policy)
 			return ret;
-		ret = cpufreq_driver_target(policy, policy->cur,
+		if (max_freq == MSM_CPUFREQ_NO_LIMIT)
+			max_freq = policy->cpuinfo.max_freq;
+		ret = cpufreq_driver_target(policy, max_freq,
 				CPUFREQ_RELATION_H);
 		cpufreq_cpu_put(policy);
 	}
+	put_online_cpus();
 
 	return ret;
 }
-- 
1.7.9.5

