From 3041cc2167527419cecb9b3b11b073d1ef1c909e Mon Sep 17 00:00:00 2001
From: Francisco Franco <franciscofranco.1990@gmail.com>
Date: Tue, 26 Nov 2013 04:38:08 +0000
Subject: [PATCH 103/517] lib: vsprintf: remove warning.

Signed-off-by: Francisco Franco <franciscofranco.1990@gmail.com>
Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 lib/vsprintf.c |    3 ---
 1 file changed, 3 deletions(-)

diff --git a/lib/vsprintf.c b/lib/vsprintf.c
index cf425cc..14d62a7 100644
--- a/lib/vsprintf.c
+++ b/lib/vsprintf.c
@@ -1719,9 +1719,6 @@ int vsnprintf(char *buf, size_t size, const char *fmt, va_list args)
 			 */
 			void *skip_arg;
 
-			WARN_ONCE(1, "Please remove ignored %%n in '%s'\n",
-					old_fmt);
-
 			skip_arg = va_arg(args, void *);
 			break;
 		}
-- 
1.7.9.5

