From da8e989f9c7995a9a231eb3b49ab7ec04f44d33e Mon Sep 17 00:00:00 2001
From: franciscofranco <franciscofranco.1990@gmail.com>
Date: Wed, 11 Dec 2013 02:15:36 +0000
Subject: [PATCH 309/517] gitignore

Signed-off-by: franciscofranco <franciscofranco.1990@gmail.com>
Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 .gitignore |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/.gitignore b/.gitignore
index 8d0ff18..0b6f47a 100644
--- a/.gitignore
+++ b/.gitignore
@@ -89,3 +89,6 @@ boot.img
 build.sh
 flash.sh
 zip/
+diff.diff
+drivers/input/touchscreen/touchboost.c
+touchboost.diff
-- 
1.7.9.5

